harmoni_actuators/harmoni_web/scripts/harmoni_web/web_service.py

```diff
@@ -70,19 +73,44 @@ def do(self, data):
         data = super().do(data)
         self.state = State.REQUEST
         self.actuation_update(actuation_completed=False)
+        data_array = self._get_web_data(data)
         try:
             rospy.sleep(1)
-            self._send_request(data)
+            for data in data_array:
+                self.send_request(data)
+                rospy.sleep(0.2)
```

Comment:
I don't know the specifics of what's going on here, but is it possible to do this without sleeps? Maybe by waiting for a response and then timing out if it takes too long?

If the sleeps are absolutely needed, then a named constant or comment would be helpful to indicate reasoning behind the duration.

Reply:
Great question, I don't know tbh @micolspitale93 would know